### PR TITLE
Note-app dev to main branch

### PR DESCRIPTION
**Pull request by mutesialine**

### CODE REVIEW

File: README.old.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# note-app
```
Comment (Jonath-z) [inline README.old.md]: You can remove this file

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+export default function App() {
+  const [notes, setNotes] = useState(
+    () => JSON.parse(localStorage.getItem("notes")) || []
+  );
+
+  const [currentNoteId, setCurrentNoteId] = React.useState(
+    (notes[0] && notes[0].id) || ""
+  );
+
+  React.useEffect(() => {
```
Comment (Jonath-z) [inline src/App.js]: Try to destructure the react hooks from the import line

File: src/App.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  function deleteNote(event, noteId) {
+    event.stopPropagation();
+    setNotes((oldNotes) => oldNotes.filter((note) => note.id !== noteId));
+  }
+
+  function findCurrentNote() {
+    return (
+      notes.find((note) => {
+        return note.id === currentNoteId;
+      }) || notes[0]
```
Comment (Jonath-z) [inline src/App.js:44]: Why this `|| notes[0]` ?

File: src/App.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+          <Sidebar
+            notes={notes}
+            currentNote={findCurrentNote()}
+            setCurrentNoteId={setCurrentNoteId}
+            newNote={createNewNote}
+            deleteNote={deleteNote}
+          />
+
+          {currentNoteId && notes.length > 0 && (
+            <Editor currentNote={findCurrentNote()} updateNote={updateNote} />
```
Comment (Jonath-z) [inline src/App.js]:
```suggestion
            <Editor currentNote={findCurrentNote} updateNote={updateNote} />
```

File: src/App.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    setCurrentNoteId(newNote.id);
+  }
+
+  return (
+    <main className="w-full max-w-7xl mx-auto">
+      {notes.length > 0 ? (
+        <Split sizes={[30, 70]} direction="horizontal" className="flex">
+          <Sidebar
+            notes={notes}
+            currentNote={findCurrentNote()}
```
Comment (Jonath-z) [inline src/App.js]:
```suggestion
            currentNote={findCurrentNote}
```

File: src/App.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
+import React from "react";
+import { useState } from "react";
```
Comment (Jonath-z) [inline src/App.js]: They can be imported on one line.
As you are using the latest version of react there is no need to explicitly import react anymore, it implicitly imported under the hood

File: src/components/Editor.jsx [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import ReactMde from "react-mde";
+import Showdown from "showdown";
+
+export default function Editor({ currentNote, updateNote }) {
+  const [selectedTab, setSelectedTab] = React.useState("write");
```
Comment (Jonath-z) [inline src/components/Editor.jsx]: here as well, you  can import destructured useState

File: src/components/Editor.jsx [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  return (
+    <section className="w-[80%] h-screen text-2xl">
+      <ReactMde
+        value={currentNote.body}
+        onChange={updateNote}
+        selectedTab={selectedTab}
+        onTabChange={setSelectedTab}
+        generateMarkdownPreview={(markdown) =>
+          Promise.resolve(converter.makeHtml(markdown))
+        }
```
Comment (Jonath-z) [inline src/components/Editor.jsx]: Try to use modern ways of creating promises in react

File: src/components/Sidebar.jsx [hunk 9]
```diff
@@ -0,0 +1,40 @@
+import React from "react";
+export default function Sidebar(props) {
+  const noteElements = props.notes.map((note, index) => (
```
Comment (Jonath-z) [inline src/components/Sidebar.jsx]:
```suggestion
  const NoteElements = props.notes.map((note, index) => (
```

File: src/components/Sidebar.jsx [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+      <div className="flex px-7  items-center gap-6 py-8">
+        <p className="text-4xl font-bold">Note</p>
+        <button
+          className="text-4xl px-5 rounded-md py-2 text-white bg-darkblue"
+          onClick={props.newNote}
+        >
+          +
+        </button>
+      </div>
+      {noteElements}
```
Comment (Jonath-z) [inline src/components/Sidebar.jsx]:
```suggestion
      <NoteElements/>
```

File: src/components/Sidebar.jsx [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+            alt="delete"
+            className="invert-0 w-fit hidden group-hover:block"
+          />
+        </button>
+      </div>
+    </div>
+  ));
+
+  return (
+    <div className=" shadow-md w-[20%] mb-16   ">
```
Comment (Jonath-z) [inline src/components/Sidebar.jsx]:
```suggestion
    <div className="shadow-md w-[20%] mb-16">
```

File: src/components/Sidebar.jsx [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+      <div className="flex px-7  items-center gap-6 py-8">
```
Comment (Jonath-z) [inline src/components/Sidebar.jsx]:
```suggestion
      <div className="flex px-7 items-center gap-6 py-8">
```

File: tailwind.config.js [hunk 13]
```diff
@@ -0,0 +1,12 @@
+/** @type {import('tailwindcss').Config} */
+module.exports = {
+  content: ["./src/**/*.{js,jsx,ts,tsx}"],
```
Comment (Jonath-z) [inline tailwind.config.js]:
```suggestion
  content: ["./src/**/*.{js,jsx}"],
```

**Review by Jonath-z — COMMENTED** [review]
@mutesialine good work overall.

File: src/App.js (hunk 3 above)
Comment (mutesialine) [reply src/App.js:44]: This code will return the note object that matches the currentNoteId state value or the first note object in the notes array if no note object matches the current currentNoteId.

**Review by Jonath-z — APPROVED** [review]

**Pull request merged**